traits/traits_listener.py

```diff
@@ -1009,9 +993,18 @@ def _get_name(self):
 
     # -- object Method Overrides ----------------------------------------------
 
-    def __init__(self, text="", **traits):
+    def __init__(self, text):
```

Comment:
im not a 100% sure why we're changing the `text` default here

Reply:
Mostly just for safety.

Previously, the `text` in the `ListenerParser` could be updated after initialization, triggering the `_text_changed` listener. With the refactored code, the only way the `ListenerParser` can be used correctly is to provide the `text` directly at initialization time. I made it non-optional so that we catch any code that tries to create a `ListenerParser` without providing the text to parse.